optimization/test_ht_functions.py

```diff
@@ -1,81 +1,109 @@
 import math
 import pytest
 from random import uniform
+import physical_properties as pp
 from ht_functions import Flow, oned_flow_modeling
 
 # parameters for test cases
+N = 6
+core_r = 0.16
 radius = 0.005
-PD = 2
+fuel_frac = 1 - (N*radius*radius*math.pi) / (core_r*core_r*math.pi)
 c = 0.00031
-L = 0.5
-N = 6
+L = core_r
+AR = 1
+power = 1e5
+T = (900, 1000)
+P = (17.9e6, 17.4e6)
+m_dot = 1
+coolant = 'CO2'
+fuel = 'UW'
+clad = 'Inconel-718'
+refl = 'Carbon'
+
+# get secondary flow properties
+flow_props = pp.FlowProperties(coolant, power, m_dot, T, P)
 
 def test_set_geom():
     """Test geometry initialization. This tests the Flow method that calculates
     flow area, fuel area and hydraulic diameter.
     """
-    test = Flow(radius, PD, c, L)
-    test.guess = N
+    test = Flow(radius, c, AR, power, fuel, coolant, clad, refl, flow_props)
+    test.core_r = core_r
+    test.fuel_frac = fuel_frac
     # expected values
     exp_De = radius * 2.0
-    exp_A_flow = math.pi * radius**2
-    exp_A_fuel = math.sqrt(3) * ((radius + c) * PD * 2)**2 / 2.0 -\
-        (radius + c)**2 * math.pi
+    exp_A_flow = (1-fuel_frac) * core_r**2 * math.pi
+    exp_A_fuel = core_r**2 * math.pi * fuel_frac
     # calculated values
     test.set_geom()
     # compare
     assert exp_De == test.D_e
-    assert exp_A_flow == test.A_flow
+    assert abs(exp_A_flow - test.A_flow) < 1e-7
     assert exp_A_fuel == test.A_fuel
 
-
 def test_characterize_flow():
     """Test flow characterization. This tests the Flow method that calculates
     important thermophysical flow characteristics like velocity, friction
     factor, and average heat-transfer coefficient.
     """
-    test = Flow(radius, PD, c, L)
-    test.guess_channels = N
+    test = Flow(radius, c, AR, power, fuel, coolant, clad, refl, flow_props)
+    test.core_r = core_r
+    test.fuel_frac = fuel_frac
     # get geom
     test.set_geom()
     # expected flow velocity
-    exp_G = test.fps.m_dot / (test.A_flow * N)
+    exp_G = test.fps.m_dot / (test.A_flow)
     exp_v = exp_G / test.fps.rho
     # expected heat transfer coefficient
     exp_Re = test.fps.rho * exp_v * test.D_e / test.fps.mu
-    exp_Nu = 0.023*math.pow(exp_Re, 0.8)*math.pow(test.fps.Pr, 0.4)
-    exp_f = 0.184 / math.pow(exp_Re, 0.2)
-    exp_h = exp_Nu * test.fps.k_cool / test.D_e
+
+    LD = L / (2*radius) 
+    relrough = test.cladprops['rough'] / (2*radius)
+    
+    # correlation take from EES
+    exp_f = 0.003030597
+    exp_Nu = 126.0544263 
+    exp_h = 1096.8527578
+    
+    print(test.D_e)
     # calculated values
     test.characterize_flow()
     # compare
-    assert exp_v == test.v
-    assert exp_f == test.f
-    assert exp_h == test.h_bar
-
+    assert abs(exp_v - test.v) < 1e-7
+    assert abs(exp_f - test.f) < 1e-7
+    assert abs(exp_Nu - test.Nu) < 1e-7
+    assert abs(exp_h - test.h_bar) < 1e-5
```

Comment:
why not `1e-7` ?

Reply:
The absolute error grows in the calculation for h_bar